### PR TITLE
New Resource: `azurerm_sentinel_data_connector_microsoft_threat_intelligence`

### PR DESCRIPTION
**Pull request by ziyeqf**
Test
---
```
terraform-provider-azurerm ❯ TF_ACC=1 go test -v ./internal/services/sentinel -run=TestAccSentinelDataConnectorMicrosoftThreatIntelligence -timeout=600m
=== RUN   TestAccSentinelDataConnectorMicrosoftThreatIntelligence_basic
=== PAUSE TestAccSentinelDataConnectorMicrosoftThreatIntelligence_basic
=== RUN   TestAccSentinelDataConnectorMicrosoftThreatIntelligence_complete
=== PAUSE TestAccSentinelDataConnectorMicrosoftThreatIntelligence_complete
=== CONT  TestAccSentinelDataConnectorMicrosoftThreatIntelligence_basic
=== CONT  TestAccSentinelDataConnectorMicrosoftThreatIntelligence_complete
--- PASS: TestAccSentinelDataConnectorMicrosoftThreatIntelligence_complete (357.80s)
--- PASS: TestAccSentinelDataConnectorMicrosoftThreatIntelligence_basic (360.50s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel      361.691s

```

### CODE REVIEW

File: internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	LookbackDate string `tfschema:"lookback_date"`
+}
+
+func (s DataConnectorMicrosoftThreatIntelligence) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```
Comment (ms-zhenhua) [inline internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go:41]: Is it limited with specific length and characters?

File: internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+			Computed:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+
+		"bing_safety_phishing_url_lookback_date": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Default:      "",
```
Comment (ms-zhenhua) [inline internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go:63]: Could this empty default value be removed?

File: internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+			ValidateFunc: validation.IsRFC3339Time,
+		},
+
+		"microsoft_emerging_threat_feed_lookback_date": {
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			Optional:     true,
+			Default:      "",
```
Comment (ms-zhenhua) [inline internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go:71]: Could this empty default value be removed?

File: internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+					DataTypes: &securityinsight.MSTIDataConnectorDataTypes{
+						BingSafetyPhishingURL:       expandSentinelDataConnectorMicrosoftThreatIntelligenceBingSafetyPhishingUrl(metaModel),
+						MicrosoftEmergingThreatFeed: expandSentinelDataConnectorMicrosoftThreatIntelligenceMicrosoftEmergingThreatFeed(metaModel),
+					},
+					TenantID: &tenantId,
+				},
+			}
+
+			// if both data types are disabled, the service will not create the data connector.
+			if dataConnector.MSTIDataConnectorProperties.DataTypes.BingSafetyPhishingURL.State == securityinsight.DataTypeStateDisabled &&
```
Comment (ms-zhenhua) [inline internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go:133]: Could this code piece be replaced by `AtLeastOneOf` in schema definition?

File: internal/services/sentinel/registration.go [hunk 5]
```diff
@@ -71,5 +71,6 @@ func (r Registration) Resources() []sdk.Resource {
 		DataConnectorOfficeIRMResource{},
 		LogAnalyticsWorkspaceOnboardResource{},
 		DataConnectorThreatIntelligenceTAXIIResource{},
+		DataConnectorMicrosoftThreatIntelligence{},
```
Comment (ms-zhenhua) [inline internal/services/sentinel/registration.go:74]: Could it be named as xxx**Resource** like other resources?

File: website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "east us"
+}
+
+resource "azurerm_log_analytics_workspace" "test" {
```
Comment (ms-zhenhua) [inline website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown:25]:
```suggestion
resource "azurerm_log_analytics_workspace" "example" {
```

File: website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+* `microsoft_emerging_threat_feed_lookback_date` - (Optional) The lookback date for the Microsoft Emerging Threat Feed in RFC3339. Changing this forces a new Data Connector to be created.
+
+-> **NOTE:** At least one of `bing_safety_phishing_url_lookback_date` and `microsoft_emerging_threat_feed_lookback_date` must be specified.
+
+---
+
+* `tenant_id` - (Optional) The ID of the tenant that this Microsoft Threat Intelligence Data Connector connects to. Changing this forces a new Microsoft Threat Intelligence Data Connector to be created.
+
+-> **NOTE** Currently, only the same tenant as the running account is allowed. Cross-tenant scenario is not supported yet.
```
Comment (ms-zhenhua) [inline website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown:67]: If Cross-tenant is not supported, maybe this property should not be exposed in current version?

Comment (ziyeqf) [reply website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown:67]: I think we can keep it to keep consistency with other data connectors?

File: internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go:41]: I have not found limit on its name..

**Review by katbyte — APPROVED** [review]
LGTM ☘️

**Pull request merged**